README.md

```diff
@@ -12,6 +12,30 @@ This is still work in progress.
 
 The blueprint contains a number of assumptions, e.g. using a monorepo using (`yarn` or `npm`) workspaces, with separate workspaces for the addon and the test-app. But there is plenty of room for bikeshedding here, so if you have suggestions about better ways to set this up, then please file an issue to discuss!
 
+### Temporary work-around for `--typescript` issues
+
+Today, `--typescript` is implemented via differing to `ember-cli`'s `--typescript` when generating the test-app.
+This has a number of issues:
+ - `ember-cli-typescript` is very out of date
+   - you'll need to remove `@types/ember__test-helpers`, `@types/ember-resolver`, and `@types/ember__string` (these provide their own types now)
+   - it's being sunset anyway, as effort is being put in to making TS support more native and less reliant on DefinitelyTyped. 
+ - peer issues with ember-cli cause the generation of a whole project to be deleted by ember-cli
+ - the package manager is ignored when ember-cli defers to `ember-cli-typescript`s blueprint (this mostly only affects `--pnpm` using projects)
+ - there is no way to opt out of ember-cli installing dependencies when using `--typescript`
+ 
+To make a monorepo manually:
+```bash
+mkdir my-monorepo
+cd my-monorepo
+git init
+touch package.json # make sure to fill this out
+touch pnpm-workspace.yaml # if you're using pnpm
+npx ember-cli@latest addon my-addon -b @embroider/addon-blueprint --skip-git --skip-npm --typescript --addon-only
+npx ember-cli@latest new test-app --skip-git --skip-npm --typescript # make sure to add the addon as a dependency in the test-app
```

Comment:
Why do you propose this as a separate step? What does it help with?

Reply:
How else do you make a v2 addon monorepo with typescript?

Reply:
Just `npx ember-cli@latest addon my-addon -b @embroider/addon-blueprint --skip-git  --typescript`?

Reply:
it's broken-by-default.

Here is that command's output in StackBlitz: https://stackblitz.com/edit/node-iuvwaw
Repro:
```
cd my-addon
yarn
```
Issues I noticed:
- using `--typescript` ended up using `--yarn`, but without teh `--yarn` flag so the v2 blueprint was generated assuming `npm` 🙈 
- `$ npm run build --workspace my-addon` seems to infinite loop under yarn 😅 
  - had to fix this / remove `prepare` to get on with this demo 😅 

I noticed that lint:types _works_ with using the non-preview types under yarn.
So here is a small change to use the preview types as in #118,
https://stackblitz.com/edit/node-okq4r6?file=my-addon%2Fpackage.json,my-addon%2Fmy-addon%2Funpublished-development-types%2Findex.d.ts,my-addon%2Fmy-addon%2Fpackage.json
```
cd my-addon
yarn
yarn lint
# error! mismatched types
```
The reason this is a yarn problem is that the v2-addon's usage of ember-source's built-in types is a _development_ concern, which the consuming app's types shouldn't care about.


The way I'm thinking of trying to solve it is to get us off of DT's types.
Part 1: https://github.com/embroider-build/addon-blueprint/pull/118
Part 2: make the app blueprint use native types as well? (or merge all the ec-typescript PRs -- but I'm not confident that the DT types can be made compatible with a yarn monorepo)

Here is an example of what the state of things are _with_ the built-in types, and also with the ec-ts changes:
https://stackblitz.com/edit/node-n5yqmy?file=my-addon%2Fpackage.json,my-addon%2Fmy-addon%2Funpublished-development-types%2Findex.d.ts,my-addon%2Ftest-app%2Fpackage.json

Same issue.

However, if the generated app were to use the built in types as well: https://stackblitz.com/edit/node-3txw9q?file=my-addon%2Fpackage.json,my-addon%2Ftest-app%2Ftypes%2Findex.d.ts,my-addon%2Ftest-app%2Fpackage.json
then everything works.

Reply:
one thing that I think could hold up the default app blueprint from using native types is that I don't know of ember-data is ready for its built in types to be used.

Reply:
If we drop support for `--yarn` then.. all of this is a non issue 😅 
(not saying we actually do that, but like all things, dropping yarn makes everything easier. lol)

Reply:
> I'd like to keep using the native types out of this discussion

They are entirely related though, to ignore native types means there is no issue with yarn.

> Again, why?

yarn is resolving types incorrectly -- afaict, is the only package manager that can't handle native types in the addon and DT types in the test app :(

Reply:
Then I don't understand the scope of this PR. It is adding a note for TS users, it is not introducing preview types. This is what #118 is doing, and we can only land that one if it works correctly, including with yarn. But without preview types, `--typescript --yarn` is (mostly) working. So what are we doing here?

> You haven't really answered that question yet. What makes running those two blueprints separately better, when our blueprint does literally the same (when not using --addon-only)

?